reference/zrc6.scilla

```diff
@@ -0,0 +1,908 @@
+      _recipient: _sender;
+      _amount: Uint128 0;
+      to: to;
+      token_id: token_id
+    };
+    msgs = one_msg msg_to_sender;
+    send msgs
+  end
+end
+
+(* Removes `to` from spender of `token_id`. *)
+(* @Requirements: *)
+(* - `token_id` should exist. Otherwise, it should throw `TokenNotFoundError` *)
+(* - `_sender` should be a token owner or an operator. Otherwise, it should throw `NotOwnerOrOperatorError` *)
+(* - `to` should be already a spender. Otherwise, it should throw `SpenderNotFoundError` *)
+transition RemoveSpender(to: ByStr20, token_id: Uint256)
+  opt_token_owner <- token_owners[token_id];
+  match opt_token_owner with
+  | None =>
+    error = TokenNotFoundError;
+    Throw error
+  | Some token_owner =>
+    RequireOwnerOrOperator token_owner;
+    
+    has_spender <- exists spenders[token_id];
+    match has_spender with
+    | True =>
+      (* Remove spender *)
+      delete spenders[token_id]
+    | False =>
+      error = SpenderNotFoundError;
+      Throw error
+    end;
+    
+    e = {
+      _eventname: "RemoveSpender";
+      to: to;
+      token_id: token_id
+    };
+    event e;
+    msg_to_sender = { 
+      _tag: "ZRC6_RemoveSpenderCallback";
+      _recipient: _sender;
+      _amount: Uint128 0;
+      to: to;
+      token_id: token_id
+    };
+    msgs = one_msg msg_to_sender;
+    send msgs
+  end
+end
+
+(* Adds `to` as operator for the `_sender`. *)
+(* @Requirements: *)
+(* - `_sender` should be the token owner. Otherwise, it should throw `NotTokenOwnerError` *)
+(* - `_sender` should not be `to`. Otherwise, it should throw `SelfError` *)
+(* - `to` should not be already an operator. Otherwise, it should throw `OperatorFoundError` *)
+transition AddOperator(to: ByStr20)
+  RequireNotSelf to _sender;
+  
+  opt_bal <- balances[_sender];
+  balance = get_bal opt_bal;
+  
+  is_balance_zero = builtin eq zero balance;
+  (* _sender should have at least 1 token *)
+  match is_balance_zero with 
+  | True =>    
+    error = NotTokenOwnerError;
+    Throw error
+  | False =>
+    has_operator <- exists operators[_sender][to];
+    match has_operator with
+    | False =>
+      (* Add operator *)
+      operators[_sender][to] := true
+    | True =>
+      error = OperatorFoundError;
+      Throw error
+    end;
+    e = {
+      _eventname: "AddOperator";
+      to: to
+    };
+    event e;
+    msg_to_sender = {
+      _tag: "ZRC6_AddOperatorCallback";
+      _recipient: _sender;
+      _amount: Uint128 0;
+      to: to
+    };
+    msgs = one_msg msg_to_sender;
+    send msgs
+  end
+end
+
+(* Removes `to` from operator for the `_sender`. *)
+(* @Requirements:  *)
+(* - `_sender` should be the token owner. Otherwise, it should throw `NotTokenOwnerError` *)
+(* - `_sender` should not be `to`. Otherwise, it should throw `SelfError` *)
+(* - `to` should be already an operator. Otherwise, it should throw `OperatorNotFoundError` *)
+transition RemoveOperator(to: ByStr20)
```

Comment:
Can consider renaming `to` to `operator_address`

Reply:
Let's go with `operator` as we discussed.